functions/src/stats.ts

```diff
@@ -0,0 +1,36 @@
+import { Lobby } from "./firestore-types/lobby";
+import type { DocumentReference, Transaction } from "firebase-admin/firestore";
+import { userCollection } from "./firestore-collections";
+import type { UserData } from "./firestore-types/users";
+
+export async function applyStats(lobbyData: Lobby, lobbyDoc: DocumentReference<Lobby>, transaction: Transaction) {
+  const { players, uids, winner } = lobbyData;
+  // update each players doc
+  await Promise.all(
+    players.map(async (player, index) => {
+      // get the player's user doc
+      const userDocRef = userCollection.doc(uids[index]);
+      const userSnapshot = await transaction.get(userDocRef);
+      const user = userSnapshot.data() as UserData;
```

Comment:
you should not be using `as` here and should be checking if it is undifined and throwing an error

Reply:
or maybe just a return and print an error message